wren-ui/src/apollo/server/repositories/modelColumnRepository.ts

```diff
@@ -114,4 +119,36 @@ export class ModelColumnRepository
       .delete();
     return await builder;
   }
+
+  public async findAllCalculatedFields(
+    queryOptions?: IQueryOptions,
+  ): Promise<ModelColumn[]> {
+    if (queryOptions && queryOptions.tx) {
+      const { tx } = queryOptions;
+      const result = await tx(this.tableName)
+        .where('is_calculated', true)
+        .select('*');
+      return result.map((r) => this.transformFromDBData(r));
+    }
+
+    const result = await this.knex<ModelColumn>('model_column')
+      .where('is_calculated', true)
+      .select('*');
+    return result.map((r) => this.transformFromDBData(r));
+  }
```

Comment:
Could we do sth like this instead ?
```js
const executer = queryOptions?.tx ? queryOptions.tx : this.knex;
```
This way you don't have to duplicate the logics.

Reply:
@onlyjackfrost said you might not need to create this method. He'll give some inputs later.

Reply:
Done by 200d415f